### ISSUE TITLE
Problem with git pull when using SSH

### ISSUE THREAD

**Issue by musicfunfan** [topgrade-rs/topgrade#875]
## Erroneous Behavior
I use OpenSSH to connect with my public repository on GitLab and GitHub. I also use a passphrase for my keys. When TopGrade starts the ``git pull`` command on the Git Repositorys with  SSH link, asks me for my passphrase. I give the passphrase but TopGrade freezes and i think that its doing nothing... 
## Expected Behavior
I should be able to provide my passphrase and then TopGrade should be able to pull from my reposotorys

## Steps to reproduce
If you use OpenSSH for Git Repo and you also use passphrase for the SSH-key try to pull repositorys using TopGrade.

## Possible Cause (Optional)
ToGrade may not be designed with passphrase in mind... ?

## Problem persists without calling from topgrade
Execute the erroneous command directly to see if the problem persists
- [x] Yes
- [ ] No

## Did you run topgrade through `Remote Execution`
- [ ] Yes
- [x] No

If yes, does the issue still occur when you run topgrade directlly in your
remote host

- [ ] Yes
- [ ] No

## Configuration file (Optional)
I do not think that it does. But my config file is hosted on [GitRepo](https://gitlab.com/musicfunfan/main_configs_files/-/blob/main/topgrade/topgrade.toml?ref_type=heads).

## Additional Details
- Operation System/Version
  ArchLinux

- Installation
  I installed the [A.U.R. package](https://aur.archlinux.org/packages/topgrade-bin).

- Topgrade version (`topgrade -V`)
Topgrade 15.0.0
## Verbose Output (`topgrade -v`)
I can not post this because it calls also a custom script i have for backup and i do not want the out put of that public. Also ``topgrade -v --disable custom_commands `` do not stop my script from executing my scripts is a ``pre_command``.

<details>
<pre>

</pre>
</details>

**Comment by musicfunfan** [comment]
Do we have something new on this matter ? Its been 3 weeks now.

**Comment by SteveLauC** [comment]
> Do we have something new on this matter ? Its been 3 weeks now.

Been really bush with my life and work for the last month, sorry about this.

Will try to take a look at this.

**Comment by musicfunfan** [comment]
No worries life changing with out asking as. Take your time no rush. If we have something on this i will get an email. Thank you for your work and time on this project.

**Comment by musicfunfan** [comment]
Do we have an update to this? 

I still can not upgrade my git repos with tograde. 😥. 

```bash
── 17:59:13 - Git repositories ─────────────────────────────────────────────────

Only updated repositories will be shown...

Failed pulling /home/musicfan/gitrepo/Hyprland
Enter passphrase for key '/home/musicfan/.ssh/glk': 
```
It ask me for my gitlab key. But first failed to pull from Hyprland repo. This the same bug as before.

**Comment by GideonBear** [comment]
This is probably caused by the concurrency on pulling git repos. Could you try setting `max_concurrency = 1` under the `[git]` section in your config file?

**Comment by musicfunfan** [comment]
Yes. i did that and now is working fine. But i do not really understand what the problem was with the set to 5. Because i think this was the default i did not set that. I just comment it out.  So if there is a problem with that we need to have the default set to 1.

**Comment by GideonBear** [comment]
> Yes. i did that and now is working fine. But i do not really understand what the problem was with the set to 5. Because i think this was the default i did not set that. I just comment it out. So if there is a problem with that we need to have the default set to 1.

Great! At least it's working for you now.
Asking input on stdin from within a thread/async doesn't go well, which is why this hangs.
I don't know how we would solve this easily... If the SSH passphrase is cached (which is pretty normal), the concurrency is fine.
